cfme/infrastructure/deployment_roles.py

```diff
@@ -71,59 +253,57 @@ def step(self):
         except NoSuchElementException:
             nav_select('Clusters / Deployment Roles')
 
-    def am_i_here(self):
-        def match(option):
-            return match_page(title=option, summary=option)
-
-        return match('Clusters / Deployment Roles') or match('Deployment Roles')
+    def resetter(self):
+        if 'Grid View' not in self.view.toolbar.view_selector.selected:
+            self.view.toolbar.view_selector.select('Grid View')
+        if self.view.paginator.exists:
+            self.view.paginator.check_all()
+            self.view.paginator.uncheck_all()
 
 
-@navigator.register(DeploymentRoles, 'AllForProvider')
-class AllForProvider(CFMENavigateStep):
-
-    def prerequisite(self):
-        navigate_to(self.obj.provider, 'Details')
-
-    def step(self):
-        def list_acc_select(option):
-            list_acc.select('Relationships', 'Show all managed {}'.format(option), by_title=True,
-                            partial=False)
+@navigator.register(DeploymentRoles, 'Details')
+class Details(CFMENavigateStep):
+    VIEW = DeploymentRoleDetailsView
+    prerequisite = NavigateToSibling('All')
 
+    def step(self, *args, **kwargs):
+        """Navigate to the details page of Role"""
         try:
-            list_acc_select('Deployment Roles')
-        except ListAccordionLinkNotFound:
-            list_acc_select('Clusters / Deployment Roles')
+            self.prerequisite_view.entities.get_entity(by_name=self.obj.name).click()
+        except:
+            raise RoleNotFound("Deployment Role {} not found".format(self.obj.name))
 
-    def am_i_here(self):
-        def match(option):
-            summary_ptrn = "{} (All {})"
-            return match_page(summary=summary_ptrn.format(self.obj.provider.name, option),
-                              title=option)
 
-        return match('Deployment Roles') or match('Clusters / Deployment Roles')
+@navigator.register(DeploymentRoles, 'ProviderDetails')
+class ProviderDetails(CFMENavigateStep):
```

Comment:
**REQUIRED**
This step is not needed. `provider` attribute of `DeploymentRoles` should have registered "Details" destination.

Reply:
yup, so thanks for correcting me. For this only I spend 1d day completely.